enterprise/reporting/event_sampler/event_sampler_test.go

```diff
@@ -0,0 +1,164 @@
+package event_sampler
+
+import (
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+)
+
+func TestBadger(t *testing.T) {
+	conf := config.New()
+	ttl := conf.GetReloadableDurationVar(500, time.Millisecond, "Reporting.eventSampling.durationInMinutes")
+	eventSamplerType := conf.GetReloadableStringVar("badger", "Reporting.eventSampling.type")
+	eventSamplingCardinality := conf.GetReloadableIntVar(10, 1, "Reporting.eventSampling.cardinality")
+	log := logger.NewLogger()
+
+	t.Run("should put and get keys", func(t *testing.T) {
+		es, _ := NewEventSampler(ttl, eventSamplerType, eventSamplingCardinality, conf, log)
+		_ = es.Put("key1")
+		_ = es.Put("key2")
+		_ = es.Put("key3")
+		val1, _ := es.Get("key1")
+		val2, _ := es.Get("key2")
+		val3, _ := es.Get("key3")
+		val4, _ := es.Get("key4")
+
+		assert.True(t, val1, "Expected key1 to be present")
+		assert.True(t, val2, "Expected key2 to be present")
+		assert.True(t, val3, "Expected key3 to be present")
+		assert.False(t, val4, "Expected key4 to not be present")
+		es.Close()
+	})
+
+	t.Run("should not get evicted keys", func(t *testing.T) {
+		es, _ := NewEventSampler(ttl, eventSamplerType, eventSamplingCardinality, conf, log)
+		_ = es.Put("key1")
+
+		time.Sleep(600 * time.Millisecond)
+		val1, _ := es.Get("key1")
+
+		assert.False(t, val1, "Expected key1 to be evicted")
+		es.Close()
+	})
+}
```

Comment:
Both these tests can fail in some cases depending if ttl runs fast/slow. I think we can use `require.Eventually`. `require.Eventually` used in multiple places in our tests